common/transport/amqp/src/amqp.ts

```diff
@@ -222,7 +188,17 @@ export class Amqp {
             this._cbs.attach(callback);
           },
           putToken: (audience, token, callback) => {
-            this._cbs.putToken(audience, token, callback);
+            if (!this._cbs) {
```

Comment:
>if (!this._cbs) { [](start = 12, length = 17)

Really this is fine, but seems a little funny that a large part of the code that was removed and replaced above was now causing an error for performing operations out of order.  But it's ok here!

Reply:
Feels as if we're missing a state. maybe we should have 'connected' -> 'initializingCBS' -> 'authenticating' -> 'authenticated'

---
In reply to: [184350713](https://github.com/Azure/azure-iot-sdk-node/pull/283#discussion_r184350713) [](ancestors = 184350713)